pkg/metrics/host.go

```diff
@@ -0,0 +1,29 @@
+// Copyright 2021 New Relic Corporation. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package metrics
+
+import (
+	"fmt"
+
+	"github.com/shirou/gopsutil/v3/host"
+)
+
+type HostSample struct {
+	Uptime uint64 `json:"uptime"`
+}
+
+type HostMonitor struct{}
+
+func NewHostMonitor() *HostMonitor {
+	return &HostMonitor{}
+}
+
+func (m *HostMonitor) Sample() (*HostSample, error) {
+	uptime, err := host.Uptime()
+	if err != nil {
+		return &HostSample{}, fmt.Errorf("cannot sample uptime: %w", err)
```

Comment:
any particular reason to initialise the empty structure for the return in error case?

Reply:
not really 🤦  changing it

Reply:
changed in 5ec5b5119c914be33d20260659dcc1b173b39039 thanks for the 👁️